etcdserver/corrupt.go

```diff
@@ -125,36 +170,59 @@ func (s *EtcdServer) checkHashKV() error {
 	return nil
 }
 
-func (s *EtcdServer) getPeerHashKVs(rev int64) (resps []*clientv3.HashKVResponse) {
-	for _, m := range s.cluster.Members() {
+type peerHashKVResp struct {
+	resp *clientv3.HashKVResponse
+	err  error
+	eps  []string
+}
+
+func (s *EtcdServer) getPeerHashKVs(rev int64) (resps []*peerHashKVResp) {
+	mbs := s.cluster.Members()
+	pURLs := make([][]string, len(mbs))
+	for _, m := range mbs {
 		if m.ID == s.ID() {
 			continue
 		}
+		pURLs = append(pURLs, m.PeerURLs)
+	}
+	if len(pURLs) == 0 {
```

Comment:
this is not needed. we never try to check hash on fresh node.

Reply:
Found that `s.cluster.Members()` is only initially populated where there's any snapshot on disk

https://github.com/coreos/etcd/blob/c983f0ae988504159c6337761027aa5ef7a3fc48/etcdserver/server.go#L389-L396

So we need a way to fallback on Raft entries from disk when we read WAL.

Reply:
if there is no snapshot, not checking the hash is fine for now.

Reply:
Ok, I will assume `RaftCluster` is populated, and change the e2e test snapshot count.